x/erc20/keeper/ibc_callbacks_test.go

```diff
@@ -2,574 +2,482 @@ package keeper_test
+			transfer: transfertypes.NewFungibleTokenPacketData(
+				dummyCoinAddress.Denom,
+				amountValue,
+				sender,
+				receiver,
+				"",
+			),
+			expectedError: nil,
 		},
-		{
-			name: "pass - denom is registered and has available balance",
-			malleate: func() transfertypes.FungibleTokenPacketData {
-				pair := suite.setupRegisterCoin(metadataIbc)
-				suite.Require().NotNil(pair)
-
-				sender := sdk.MustAccAddressFromBech32(senderAddr)
-
-				// Mint coins on account to simulate receiving ibc transfer
-				coinEvmos := sdk.NewCoin(pair.Denom, math.NewInt(10))
-				coins := sdk.NewCoins(coinEvmos)
-				err := suite.app.BankKeeper.MintCoins(suite.ctx, inflationtypes.ModuleName, coins)
-				suite.Require().NoError(err)
-				err = suite.app.BankKeeper.SendCoinsFromModuleToAccount(suite.ctx, inflationtypes.ModuleName, sender, coins)
-				suite.Require().NoError(err)
 
-				return transfertypes.NewFungibleTokenPacketData(pair.Denom, "10", senderAddr, "", "")
-			},
-			expPass: true,
+		{
+			name: "no-op - is native Erc20 - sender is module address",
+			transfer: transfertypes.NewFungibleTokenPacketData(
+				erc20TestPair.Denom,
+				amountValue,
+				unitNetwork.App.AccountKeeper.GetModuleAddress("erc20").String(),
+				receiver,
+				"",
+			),
+			expectedError: nil,
 		},
+		// Erc20 is registered but there is no balance on the account
+		// It should error when attempting to convert coin -> erc20
 		{
-			name: "error - denom is registered but has no available balance",
-			malleate: func() transfertypes.FungibleTokenPacketData {
-				pair := suite.setupRegisterCoin(metadataIbc)
-				suite.Require().NotNil(pair)
-
-				return transfertypes.NewFungibleTokenPacketData(pair.Denom, "10", senderAddr, "", "")
-			},
-			expPass: false,
+			name: "error - is native Erc20 - no coin balance for convertion",
+			transfer: transfertypes.NewFungibleTokenPacketData(
+				erc20TestPair.Denom,
+				amountValue,
+				sender,
+				receiver,
+				"",
+			),
+			expectedError: errorsmod.Wrap(types.ErrEVMCall, "failed to retrieve balance"),
 		},
 	}
 	for _, tc := range testCases {
-		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
-			suite.mintFeeCollector = true
-			suite.SetupTest() // reset
-
-			transfer := tc.malleate()
+		suite.Run(tc.name, func() {
+			err := unitNetwork.App.Erc20Keeper.ConvertCoinToERC20FromPacket(
+				unitNetwork.GetContext(),
+				tc.transfer,
+			)
 
-			err := suite.app.Erc20Keeper.ConvertCoinToERC20FromPacket(suite.ctx, transfer)
-			if tc.expPass {
-				suite.Require().NoError(err)
+			if tc.expectedError != nil {
+				suite.Require().Equal(tc.expectedError.Error(), err.Error())
 			} else {
-				suite.Require().Error(err)
+				suite.Require().Equal(tc.expectedError, err)
 			}
+
 		})
 	}
 }
 
-func (suite *KeeperTestSuite) TestOnAcknowledgementPacket() {
-	var (
-		data transfertypes.FungibleTokenPacketData
-		ack  channeltypes.Acknowledgement
-		pair *types.TokenPair
+// TODO: coverage? or delete?
```

Comment:
address this TODO

Reply:
This was more of a question.
Do we add simple tests to comply with code coverage or do we add the comment to ignore coverage on the function?